### PR TITLE
Issue 1618 limit accepted message passwords based on password policy

### LINKED ISSUE #1618: limit accepted message passwords based on password policy
part of #1545

### PR DESCRIPTION
**Pull request by DenBond7**
This PR added password policy

close #1618

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities

### CODE REVIEW

File: FlowCrypt/src/main/java/com/flowcrypt/email/ui/activity/fragment/CreateMessageFragment.kt [hunk 1]
```diff
@@ -195,6 +195,11 @@ class CreateMessageFragment : BaseSyncFragment(), View.OnFocusChangeListener,
     }
   }
 
+  override fun onDestroyView() {
+    super.onDestroyView()
```
Comment (IvanPizhenko) [inline FlowCrypt/src/main/java/com/flowcrypt/email/ui/activity/fragment/CreateMessageFragment.kt:199]: If in onCreate() the sequence is such we first run superclass stuff then local, then, logically, destruction should do things in the opposite way. But here I see that `super.onDestroyView()` is called first, then some local cleanup is performed. Is it correct? If correct then why?

File: FlowCrypt/src/main/java/com/flowcrypt/email/ui/activity/fragment/CreateMessageFragment.kt [hunk 2]
```diff
@@ -195,6 +195,11 @@ class CreateMessageFragment : BaseSyncFragment(), View.OnFocusChangeListener,
...
+    appBarLayout?.removeView(nonEncryptedHintView)
+  }
+
   override fun onDestroy() {
     super.onDestroy()
```
Comment (IvanPizhenko) [inline FlowCrypt/src/main/java/com/flowcrypt/email/ui/activity/fragment/CreateMessageFragment.kt:204]: same question as above - why `super.onDestroy()` call comes first?

**Review by IvanPizhenko — COMMENTED** [review]
I have couple of questions here

File: FlowCrypt/src/main/java/com/flowcrypt/email/ui/activity/fragment/CreateMessageFragment.kt (hunk 1 above)
Comment (DenBond7) [reply FlowCrypt/src/main/java/com/flowcrypt/email/ui/activity/fragment/CreateMessageFragment.kt:199]: :) It refers to the Fragments lifecycle. Usually, in overridden methods, we should call the super methods first to perform the basic logic that is described in the superclass. Sorry for the short answer but it's a little difficult to explain how Android works in a few lines :)

Comment (DenBond7) [reply FlowCrypt/src/main/java/com/flowcrypt/email/ui/activity/fragment/CreateMessageFragment.kt:199]: You can read more info here https://developer.android.com/guide/components/activities/activity-lifecycle

**Review by IvanPizhenko — APPROVED** [review]

**Pull request merged**